### ISSUE TITLE
Official Godot binary won't run on several devices returning the same error. (32-bit Linux)

### ISSUE THREAD

**Issue by DegradedMind** [godotengine/godot#27496]
**Godot version:**
3.1-stable - 3.1.2 (Not Mono)
`
Same issue here:
#27226 

**OS/device including version:**
1. Dell Inspirion 6400
```
~$ neofetch
ubuntu@lubuntu
---------------
OS: lubuntu 16.04 32-bit i686
Host: Inspirion 6400
Kernel: 4.13.0-16-generic
Resolution: 1280x800
DE: LXQT
CPU: Genuine Intel T2500 (2) @ 2.000GHz
GPU: AMD Mobility Radeon X1300
Memory: 324MiB / 6022MiB
```
```
~$ glxinfo | grep "OpenGL"
OpenGL vendor string: X.Org R300 Project
OpenGL renderer string: ATI RV515
OpenGL version string: 2.1 Mesa 17.2.2
OpenGL shading language version string: 1.20
OpenGL ES profile version string: OpenGL ES 2.0 Mesa 17.2.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 1.0.16
```
2. HP Pavilion DV5
```
~$ neofetch 
lubuntu@lubuntu
---------------
OS: lubuntu 17.04 32bit i686
Host: HP Pavilion dv5 Notebook PC Rev 1
Kernel: 4.13.0-16-generic
Resolution: 1280x800
DE: LXDE
CPU: Intel 2 Duo P8400 (2) @ 2.267GHz
GPU: NVIDIA GeForce 9200M GS
Memory: 1203MiB / 4007MiB
```
```
~$ glxinfo | grep "OpenGL"
OpenGL core profile version string: 3.3 (Core Profile) Mesa 17.2.2
OpenGL core profile shading language version string: 3.30
OpenGL version string: 3.0 Mesa 17.2.2
OpenGL shading language version string: 1.30
OpenGL ES profile version string: OpenGL ES 3.0 Mesa 17.2.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
```
3. Lenovo Thinkpad T60
```
~$ neofetch
live@debian 
----------- 
OS: Debian GNU/Linux 9.0 (stretch) i686 
Model: 20074AU ThinkPad T60 
Kernel: 4.9.0-3-686 
Resolution: 1024x768 
DE: XFCE
CPU: Intel 2 T7400 (2) @ 2.1GHz 
GPU: AMD Mobility Radeon X1300 
Memory: 776MB / 3033MB
``` 
```
~$ glxinfo | grep "OpenGL"
OpenGL vendor string: X.Org R300 Project
OpenGL renderer string: Gallium 0.4 on ATI RV515
OpenGL version string: 2.1 Mesa 13.0.6
OpenGL shading language version string: 1.20
OpenGL ES profile version string: OpenGL ES 2.0 Mesa 13.0.6
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 1.0.16
```

**Issue description:**
Official 32-bit Godot binary returns the same error on several devices :

```
~$ ./Godot_v3.1-stable_x11.32
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] linux-gate.so.1(__kernel_sigreturn+0) [0xb772dce4] (??:0)
-- END OF BACKTRACE --
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] linux-gate.so.1(__kernel_sigreturn+0) [0xb772dce4] (??:0)
-- END OF BACKTRACE --
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] linux-gate.so.1(__kernel_sigreturn+0) [0xb772dce4] (??:0)
-- END OF BACKTRACE --
Aborted
```

Running with `--video-driver GLES2 --verbose` doesn't change anything.
But the `3.1-stable` binary compiled with `scons p=x11` from the source runs fine.

**Comment by akien-mga** [comment]
CC @hpvb. Might be due to LTO. If so it's probably not a big loss to disable it on 32-bit x86.

**Comment by qarmin** [comment]
In my Lubuntu 18.04.2 i686 with Intel HD 3000 and Mesa 18.2.2 core profile 3.3. Linux 4.18.0-15 everything works with only this error(probably because I use Live version)
```
ERROR: set_locale: Unsupported locale 'C', falling back to 'en'.
```

**Comment by DegradedMind** [comment]
Live Lubuntu 18.10 i686 with Intel Core2 Intel T2500 - the issue still present.

**Comment by sackondeeznuts1** [comment]
Can confirm the issue: on my 64-bit system the `.32` Godot executable returns the same error:
```
$ ./Godot_v3.1-stable_x11.32
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] linux-gate.so.1(__kernel_sigreturn+0) [0xf7f9ce10] (??:0)
-- END OF BACKTRACE --
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] linux-gate.so.1(__kernel_sigreturn+0) [0xf7f9ce10] (??:0)
-- END OF BACKTRACE --
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] linux-gate.so.1(__kernel_sigreturn+0) [0xf7f9ce10] (??:0)
-- END OF BACKTRACE --
Aborted (core dumped)
```
My system:
```
$ neofetch
manjaro@manjaro 
--------------- 
OS: Manjaro Linux x86_64 
Host: Vostro 1500 
Kernel: 4.19.28-1-MANJARO 
Uptime: 9 mins 
Packages: 1169 (pacman) 
Shell: bash 5.0.0 
Resolution: 1680x1050 
DE: Xfce 
WM: Xfwm4 
WM Theme: Adapta-Maia 
Theme: Adapta-Eta-Maia [GTK2], Adwaita [GTK3] 
Icons: Papirus-Maia [GTK2], Adwaita [GTK3] 
Terminal: xfce4-terminal 
Terminal Font: Monospace 12 
CPU: Intel Core 2 Duo T8100 (2) @ 2.101GHz 
PU: NVIDIA GeForce 8400M GS 
Memory: 1469MiB / 5999MiB 
```
```
$ glxinfo | grep "OpenGL"
OpenGL vendor string: nouveau
OpenGL renderer string: NV86
OpenGL core profile version string: 3.3 (Core Profile) Mesa 18.3.4
OpenGL core profile shading language version string: 3.30
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.3 (Compatibility Profile) Mesa 18.3.4
OpenGL shading language version string: 3.30
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.0 Mesa 18.3.4
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
OpenGL ES profile extensions:
```

**Comment by qarmin** [comment]
Can reproduce on Lubuntu 18.04.2 32 on GTX 970 with nouveau driver
I can't compile Godot with use_lto=yes option because it show this errors:
```
/tmp/ccF3yayF.ltrans28.ltrans.o:(.data.rel.local+0x6540): undefined reference to `.L28266'
/tmp/ccF3yayF.ltrans28.ltrans.o:(.data.rel.local+0x6544): undefined reference to `.L28267'
/tmp/ccF3yayF.ltrans28.ltrans.o:(.data.rel.local+0x6548): undefined reference to `.L28268'
```
with use_lto=no everything seems to work

**Comment by DegradedMind** [comment]
There is something about the issue right there: https://github.com/BlindMindStudios/StarRuler2-Source/issues/24

**Comment by marxin** [comment]
I can take a look tomorrow.

**Comment by hpvb** [comment]
I will try to reproduce. I haven't actually tried to run the 32bit linux binaries in a while.

**Comment by JeremiahLR** [comment]
Using Lubuntu 18. Issue still present. Any workarounds?

**Comment by Zireael07** [comment]
@JeremiahLR: Tried 3.1 or 3.1.1?

**Comment by JeremiahLR** [comment]
It was 3.1.1. But you know what, I just downloaded the Mono Version with C# Support and it works on my Lubuntu! Great!

**Comment by akien-mga** [comment]
Does it work with `DRI_PRIME=0 ./Godot_v3.1-stable_x11.32`?

**Comment by akien-mga** [comment]
Can anyone test if the issue is still reproducible with 3.1.2 or the latest 3.2 build?

Also with `DRI_PRIME=0` as mentioned above.

**Comment by DegradedMind** [comment]
The issue is still present in 3.1.2 even with `DRI_PRIME=0`. With `DRI_PRIME=0` the error is printed only once:

```
$ DRI_PRIME=0 ./Godot_v3.1.2-stable_x11.32 
Godot Engine v3.1.2.stable.official - https://godotengine.org
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] [0xb7f30d10] (??:0)
-- END OF BACKTRACE --
Aborted (core dumped)
```

instead of three:

```
$ ./Godot_v3.1.2-stable_x11.32 
Godot Engine v3.1.2.stable.official - https://godotengine.org
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] [0xb7f58d10] (??:0)
-- END OF BACKTRACE --
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] [0xb7f58d10] (??:0)
-- END OF BACKTRACE --
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] [0xb7f58d10] (??:0)
-- END OF BACKTRACE --
Aborted (core dumped)

```

**Comment by DegradedMind** [comment]
Mono builds work fine.

**Comment by akien-mga** [comment]
Duplicate of #27496.

3.2 RC 1 should work fine (at least it does for me on Ubuntu Studio 18.04 i386).
